frontend/public/components/utils/webhooks.tsx

```diff
@@ -90,27 +133,45 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
     }
 
     const secretName = _.get(trigger, [triggerProperty, 'secretReference', 'name']);
-    k8sGet(SecretModel, secretName, namespace).then(
-      (secret: K8sResourceKind) => {
-        if (!_.has(secret, 'data.WebHookSecretKey')) {
-          errorModal({
-            error: `Secret referenced in the ${triggerProperty} webhook trigger does not contain 'WebHookSecretKey' key. Webhook trigger won’t work due to the invalid secret reference`,
-          });
-          return;
-        }
-        const webhookSecretValue = Base64.decode(secret.data.WebHookSecretKey);
-        const url = getWebhookURL(trigger, webhookSecretValue);
-        navigator.clipboard.writeText(url);
-      },
-      (err) => {
-        const error = err.message;
-        errorModal({ error });
-      },
+    const webhookSecret: K8sResourceKind = webhookSecrets.find(
+      (secret: K8sResourceKind) => secret.metadata.name === secretName,
+    );
+    if (!_.has(webhookSecret, 'data.WebHookSecretKey')) {
+      errorModal({
+        error: `Secret referenced in the ${triggerProperty} webhook trigger does not contain 'WebHookSecretKey' key. Webhook trigger won’t work due to the invalid secret reference`,
+      });
+      return;
+    }
+    const webhookSecretValue = Base64.decode(webhookSecret.data.WebHookSecretKey);
+    const url = getWebhookURL(trigger, webhookSecretValue);
+    navigator.clipboard.writeText(url);
+  };
+
+  const getClipboardButton = (trigger: WebhookTrigger) => {
+    const triggerProperty = getTriggerProperty(trigger);
+    const plainSecret = _.get(trigger, [triggerProperty, 'secret']);
+    const secretReference = _.get(trigger, [triggerProperty, 'secretReference', 'name']);
+    const webhookSecret: K8sResourceKind = webhookSecrets.find(
+      (secret: K8sResourceKind) => secret.metadata.name === secretReference,
     );
+    return webhookSecret || plainSecret ? (
+      <Button variant="link" type="button" onClick={() => copyWebhookToClipboard(trigger)}>
+        <PasteIcon />
+        &nbsp;Copy URL with Secret
+      </Button>
+    ) : null;
   };
 
   return (
     <div className="co-m-pane__body">
+      {!_.isEmpty(secretErrors) && (
+        <ExpandableAlert
+          variant={AlertVariant.warning}
+          alerts={_.map(secretErrors, (error, i) => (
+            <React.Fragment key={i}>{error}</React.Fragment>
```

Comment:
Should this be a `div` so each alert is on its own line?

```suggestion
            <div className="co-pre-line" key={i}>{error}</div>
```